### PR TITLE
Directly use network provider to await for transaction completion

### PR DESCRIPTION
**Pull request by popenta**
Removed the `ProxyProviderWrapper` as it is not needed because the network providers have the capability to `await_transaction_completion`.

Also, removed the `SmartContractQueryConverter` as it did not simply unwrap the object to a dictionary, but to a specific dictionary accepted by the Api. Placed the logic in a new file called `http_resources`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 jobs:
   coverage:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.11]
+      max-parallel: 1  # This ensures jobs run sequentially, not concurrently
```
Comment (popenta) [inline .github/workflows/ci.yml:16]: By running the tests sequentially, we eliminate the occasional nonce collision. Sometimes, we used to get `lower nonce in tx`.

File: multiversx_sdk/network_providers/http_resources.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def smart_contract_query_to_dictionary(query: SmartContractQuery) -> Dict[str, Any]:
+    request: Dict[str, Any] = {
+        'scAddress': query.contract,
+        'funcName': query.function,
+        'value': str(query.value if query.value else 0),
+        'args': [arg.hex() for arg in query.arguments]
+    }
+    caller = query.caller
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/http_resources.py:15]: Can be moved under `if`.

File: multiversx_sdk/network_providers/http_resources.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    if caller:
+        request['caller'] = caller
+
+    return request
+
+
+def raw_query_response_to_smart_contract_query_response(raw_response: Dict[str, Any], function: str) -> SmartContractQueryResponse:
```
Comment (andreibancioiu) [inline multiversx_sdk/network_providers/http_resources.py:23]: Maybe `raw_vmquery_...` or `raw_contract_query_...` (more explicit)?

**Review by danielailie — DISMISSED** [review]

File: multiversx_sdk/network_providers/http_resources.py (hunk 2 above)
Comment (andreibancioiu) [reply multiversx_sdk/network_providers/http_resources.py:15]: Optionally, let's think of a naming convention for functions in this file. E.g.:

- `smart_contract_query_to_vm_values_query_request` 
- `vm_values_query_response_to_smart_contract_query_response`

(just an idea).

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk/network_providers/http_resources.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk/network_providers/http_resources.py:15]: Removed the `caller` variable as it was unnecessary. Also renamed the functions.

**Review by andreibancioiu — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**